examples/interrupt.c

```diff
@@ -0,0 +1,141 @@
+/*
+Example of instantiating of the WebAssembly module and invoking its exported
+function.
+
+You can compile and run this example on Linux with:
+
+   cargo build --release -p wasmtime
+   cc examples/interrupt.c \
+       -I crates/c-api/include \
+       -I crates/c-api/wasm-c-api/include \
+       target/release/libwasmtime.a \
+       -lpthread -ldl -lm \
+       -o interrupt
+   ./interrupt
+
+Note that on Windows and macOS the command will be similar, but you'll need
+to tweak the `-lpthread` and such annotations as well as the name of the
+`libwasmtime.a` file on Windows.
```

Comment:
Also add a test for this file over here: https://github.com/bytecodealliance/wasmtime/blob/master/crates/c-api/tests/wasm-c-examples.rs

Reply:
Hm I don't think this is actually that necessary, we already run all the examples on CI with `cargo run -p run-examples`?

Reply:
Ah ok, I guess https://github.com/bytecodealliance/wasmtime/pull/1463 was unnecessary